fendermint/app/src/app.rs

```diff
@@ -291,6 +299,37 @@ where
         let state = self.committed_state()?;
         Ok((state.state_params, state.block_height))
     }
+
+    fn verify_ipc_message(&self, ipc_message: IPCMessage) -> bool {
```

Comment:
This can now be moved into the `ChainMessageInterpreter` in the implementation of the `ProposalInterpreter` added in https://github.com/consensus-shipyard/fendermint/pull/185

My goal with `app.rs` was to keep its size and scope to a minimum, it's arguably too big already. Ideally it should only be an implementation of the `Application` trait and dispatch to other components. Everything it does is in support of this, mainly state management and conversions to Tendermint format. 

It is for this reason that the top-level interpreter which deals with transactions and genesis receives binary _input_: to make `app.rs` oblivious to the actual payloads we are working with. It has to know the _output_ format in order to be able to translate it to Tendermint types. 

I'd like us to maintain this separation of concerns and move the finality checks into the interpreter. This can be achieved in multiple ways:

The _parent view_ or `parent_finality` can be added to the `ChainMessageInterpreter` during application construction, instead of the `App` struct, but ideally only if it's a _stateless_ component. If parent finality checking needs some state, it should be lifted into the `App` for safekeeping, and passed to the interpreter during invocation. I would try using STM for this, so the interpreter can read the state of the parent chain from data structures maintained by another background component, which can be started `run.rs`. This same STM enabled componet can also be a shared with the transaction processing where we can add things CIDs to look up, for another background process to handle.